packages/redux-query-react/src/hooks/use-request-on-demand.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+// NOTE(kelson): The public flow type interface for this hook is defined in use-request-on-demand.js.flow
+
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { requestAsync } from 'redux-query';
+import type { ActionPromiseValue, QueryConfig } from 'redux-query/types.js.flow';
+
+import type { QueryState } from '../types';
+import useQueryState from './use-query-state';
+
+const useRequestOnDemand = (
+  makeQueryConfig: (...args: $ReadOnlyArray<mixed>) => QueryConfig,
```

Comment:
Just wondering why you chose to pass the query config creator here instead of the created query config?

Reply:
was following existing patterns + this allows us to change the query config on subsequent requests by passing arguments to the returned callback

Reply:
Yeah I think this is one of those things that initially looks weird but leads to much better ergonomics.